Felienne/hedy
# [FEATURE] Implements stop button

**Description**
In this PR we proudly present: a stop button! Currently users are unable to stop programs at any time they want, the only option so to run another program to "overwrite" the current running program. However, this in turn messed with a possible `Promise` created by the inline-modal waiting for input. This prevent this created a bucket-fix to disable to run button as long as there is a modal waiting. Not ideal!

We fix all these issues in this PR. We implement a stop button that shows directly when a program starts running. A user can press this button anything to terminate the program, also when there is a waiting inline-modal. We implemented a work-around by faking some input and directly hiding the program, making it seems as nothing has happened.

**Fixes**
This PR fixes #2570.

**How to test**
Test a variety of programs containing sleep, repeat and possible infinity loops. Verify that the stop button shows as soon as you press the run button and also verify that it terminates the programs successfully in all situations.
